tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,84 @@
+from marionette.wait import Wait
+
+from gaiatest.apps.settings.app import Settings
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+        """
+        https: // bugzilla.mozilla.org / show_bug.cgi?id = 1118054
+        """
+
+        self.settings = Settings(self.marionette)
+        self.settings.launch()
+        datetime = self.settings.open_date_and_time_settings()
+
+        # Auto time update is by default set to true, turn it off to make region change
+        datetime.toggle_automatic_time_update()
+        self.assertFalse(datetime.is_autotime_enabled, 'Autotime still enabled')
+
+        # record time and change the region
+        old_time = datetime.get_current_time_text
+        datetime.set_region('Atlantic Ocean')
+
+        # get the displayed time after the region change
+        new_time = datetime.get_current_time_text
+        self.assertNotEqual(new_time, old_time)
+
+        # lock screen and check time on the lockscreen
+        self.marionette.switch_to_frame()
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+        self.assertLessEqual(self.get_time_difference(new_time, lock_screen.time), 1)
+
+        # configure to set the time automatically (this will revert the timezone change), then lock screen
+        lock_screen.switch_to_frame()
+        lock_screen.unlock()
+        self.apps.switch_to_displayed_app()
+
+        # Enable the auto time update, so the regions change back and date/time is reverted back
+        datetime.toggle_automatic_time_update()
+        self.assertTrue(datetime.is_autotime_enabled, 'Autotime still disabled')
+        self.marionette.switch_to_frame()
+        self.device.lock()
+
+        # wait until device is off and turn back on to check that the time is changed
+        Wait(self.marionette, timeout=20).until(
+            lambda m: not self.device.is_screen_enabled)
+        self.device.turn_screen_on()
+
+        # Check it reverted to the correct time, and compare it with the previously shown time
+        # Allow 5 minutes difference max
+        self.assertLessEqual(self.get_time_difference(old_time, lock_screen.time), 5)
+
+    def get_time_difference(self, old_time, new_lockscreen_time):
```

Comment:
ah, i was looking for this comment, and thought you deleted them.  the thing is that I still need the roundabout way of computing the difference, because in case new_time = 1:01 and lockscreen_time = 12:59, the difference is 2 min, but simple subtraction would give me 11:58.  Hence I have to convert them to minutes, check whether it's 12 o'clock, and do subtraction.  In order to do this, I find it simpler for me to extract the numbers from the string, rather than using the time format (well, at least easier to see the logic when you read)

This is all because the lockscreen won't display AM/PM.  very annoying.  actually, it should be a bug really.